core/src/test/java/com/linecorp/armeria/client/AdditionalAuthorityTest.java

```
@@ -0,0 +1,184 @@
+        serverAPort = serverA.httpPort();
+    }
+
+    @AfterAll
+    static void afterAll() {
+        client.options().factory().closeAsync();
+    }
+
+    @Test
+    void additionalAuthorityHasHighestPrecedent() {
+        try (SafeCloseable ignored = Clients.withContextCustomizer(
+                ctx -> ctx.addAdditionalRequestHeader(HttpHeaderNames.AUTHORITY,
+                                                      "bar:" + serverAPort))) {
+
+            assertThat(client.get("http://foo:" + serverAPort).aggregate().join().contentUtf8())
+                    .isEqualTo("bar/bar:" + serverAPort);
+        }
+    }
+
+    @Test
+    void requestHeader() {
+        final HttpRequest request = HttpRequest.of(RequestHeaders.builder(HttpMethod.GET, "/")
+                                                                 .scheme("http")
+                                                                 .authority("bar:" + serverAPort)
+                                                                 .build());
+        assertThat(client.execute(request).aggregate().join().contentUtf8())
+                .isEqualTo("bar/bar:" + serverAPort);
+    }
+
+    @Test
+    void noAuthority() {
+        final HttpRequest request = HttpRequest.of(RequestHeaders.of(HttpMethod.GET, "/"));
+        assertThatThrownBy(() -> client.execute(request).aggregate().join())
+                .isInstanceOf(CompletionException.class)
+                .hasCauseInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining("Scheme and authority must be specified in");
+    }
+
+    @Test
+    void absolutePath() {
+        final HttpRequest request =
+                HttpRequest.of(RequestHeaders.of(HttpMethod.GET, "http://foo:" + serverAPort));
+        assertThat(client.execute(request).aggregate().join().contentUtf8()).isEqualTo(
+                "foo/foo:" + serverAPort);
+    }
+
+    @Test
+    void shouldIgnoreInvalidAdditionalAuthority() {
```

Should we fail the request instead of ignoring an invalid value?

---

I followed the original policy but it makes sense to fail the request. Let me address the comment.